From 384c33c1293f97a2b85c782ec468bd3ec2634483 Mon Sep 17 00:00:00 2001
From: jinghua <jinghua@marvell.com>
Date: Thu, 28 Apr 2016 22:27:43 +0800
Subject: [PATCH 0523/2241] fix: dts: a3700: add pinctrl and gpio property for
 pci reset

The pinctrl and gpio DT attributes for PCIe are added
in order to support resetting PCIe EP by GPIO.

Change-Id: Ic2a24b5927f12004b24d3bca01e55ba9ed07a212
Signed-off-by: jinghua <jinghua@marvell.com>
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29628
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
Verified-Armada37x0: Wilson Ding <dingwei@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts | 5 +++++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index fde3fdd..ae07487 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -177,6 +177,11 @@
 			status = "okay";
 
 			pcie@1,0 {
+				reset-gpios = <&gpio_sb 3 GPIO_ACTIVE_LOW>;
+
+				pinctrl-names = "default";
+				pinctrl-0 = <&pcie_pins>;
+
 				status = "okay";
 			};
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index f560582..f0a7768 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -187,6 +187,11 @@
 					marvell,pins = "mpp24", "mpp25", "mpp26", "mpp27", "mpp28", "mpp29";
 					marvell,function = "sdio";
 				};
+
+				pcie_pins: pcie_pins {
+					marvell,pins = "mpp3";
+					marvell,function = "gpio";
+				};
 			};
 
 			cpufreq@14000 {
-- 
2.7.4

